pydicom/dataset.py

```diff
@@ -179,6 +179,19 @@ def add_new(self, element_offset, VR, value):
         self.dataset.add_new(self.get_tag(element_offset), VR, value)
 
 
+def _dict_equal(a, b, exclude=[]):
```

Comment:
use `None` instead of a (mutable) list as default parameter

Reply:
I'm aware of the dangers of passing lists as default args.  Probably should change it, but here it is empty and never written to, so thought it would be okay...  Alternatively, could just take it out and always exclude `_dict` - it won't hurt Datasets anyway.

Reply:
Well, it would put `_dict` into the `exclude` parameter the first time it is called with that parameter - so basically it would be called with that parameter always after that. So if you just always exclude `_dict` it wouldn't change the behavior anyway...

Reply:
I changed it anyway, it is easier to understand with None.  At first I just blanked on the proper list comprehension so that I didn't have to split the boolean logic into separate lines.  Got it now, pushing it up shortly.